atom/browser/api/atom_api_web_contents.cc

```diff
@@ -2576,13 +2616,31 @@ void WebContents::OnTabCreated(const mate::Dictionary& options,
     }
   }
 
-  int opener_tab_id = -1;
+  int opener_tab_id = TabStripModel::kNoTab;
   options.Get("openerTabId", &opener_tab_id);
 
   content::WebContents* source = nullptr;
-  if (opener_tab_id != -1) {
+  if (opener_tab_id != TabStripModel::kNoTab) {
     source = extensions::TabHelper::GetTabById(opener_tab_id);
     tab_helper->SetOpener(opener_tab_id);
+    if (owner_window() && owner_window()->browser()) {
```

Comment:
also, isn't this mostly the same code from `TabStripModelOrderController::DetermineInsertionIndex` ? Can we call that instead?